server/operations.js

```diff
@@ -5,38 +5,43 @@ File that stores neighborhoodList in memory. Performs all CRUD operations on nei
 */
 
 const csv = require('./csv')
+
+let isLoaded = false
 let neighborhoodList = []
+class OperationsLayer {
+	//Singleton constructor. If data already loaded, then return the list. Otherwise, load the list.
+	constructor() {
+		if(!isLoaded) {
+			neighborhoodList = csv.load()
+			isLoaded = true
+		}
+		return neighborhoodList
+	}
 
-/*
+	/*
 	Function to get the entire NeighborhoodList array from the csv.
-	TODO(?): Figure out a way to store the neighborhoodList in memory so that we don't have to parse the csv file
-		every time we want to access data. load() function doesn't take that much time, so might not be worth the effort.
-*/
-function getNeighborhoodList() {
-	neighborhoodList = csv.load()
-	return neighborhoodList
-}
+	*/
+	static getNeighborhoodList() {
+		return neighborhoodList
+	}
 
-/*
+	/*
 	Test function to see filtering in action
-*/
-function filterByAll(constraintArray) {
-	if(neighborhoodList.length == 0) {
-		getNeighborhoodList();
+	*/
+	static filterByAll(constraintArray) {
+		if(!isLoaded) initializeDataLayer();
+		// console.log(constraintArray)
+		const newList = neighborhoodList
+		.filter( element => element.median_value >= constraintArray.minMedianHousePrice)
+		.filter( element => element.median_value <= constraintArray.maxMedianHousePrice)
+		.filter( element => element.latitude >= constraintArray.minLatitude)
+		.filter( element => element.latitude <= constraintArray.maxLatitude)
+		.filter( element => element.longitude >= constraintArray.minLongitude)
+		.filter( element => element.longitude <= constraintArray.maxLongitude)
+		return newList
 	}

```

Comment:
I want to create a new file call analytics.js that calls the filterByAll() function with different constraintArray constraints. We can construct the constraint array using data we send from client GET requests.